vaadinfrontend/src/main/java/life/qbic/datamanager/views/projects/project/experiments/experiment/components/ExperimentalGroupsDialog.java

```diff
@@ -0,0 +1,263 @@
+      groupEntry.setAvailableVariableLevels(experimentalVariables);
+      experimentalGroupsCollection.add(groupEntry);
+      groupEntry.registerForRemoveEvents(event -> removeExperimentalGroupEntry(event.getSource()));
+    }
+  }
+
+  /**
+   * Provides the current experimental groups defined by the user.
+   *
+   * @return a collection of {@link ExperimentalGroupContent}, describing the group size (biological
+   * replicates) and the variable level combination.
+   * @since 1.0.0
+   */
+  public Collection<ExperimentalGroupContent> experimentalGroups() {
+    return this.experimentalGroupsCollection.getChildren()
+        .filter(component -> component.getClass().equals(ExperimentalGroupEntry.class))
+        .map(experimentalGroupEntry -> convert((ExperimentalGroupEntry) experimentalGroupEntry))
+        .toList();
+  }
+
+  public record ExperimentalGroupContent(int size, Collection<VariableLevel> variableLevels) {}
+
+  class ExperimentalGroupEntry extends Div {
+
+    @Serial
+    private static final long serialVersionUID = -1387021927263833261L;
+    private final List<ComponentEventListener<RemoveEvent>> removeEventListeners = new ArrayList<>();
+    private final MultiSelectComboBox<VariableLevel> condition = new MultiSelectComboBox<>();
+    @Min(1)
+    private final NumberField sampleSize = new NumberField();
+
+    private ExperimentalGroupEntry() {
+      super();
+      addClassName("experimental-group-entry");
+
+      configureNumberFied();
+      configureConditionField();
+
+      var deleteIcon = new Icon(VaadinIcon.CLOSE_SMALL);
+      deleteIcon.addClickListener(listener -> notifyListenersAboutRemove());
+
+      add(condition, sampleSize, deleteIcon);
+    }
+
+    private void configureConditionField() {
+      condition.setItems(new ArrayList<>());
+      condition.setItemLabelGenerator(VARIABLE_LEVEL_ITEM_LABEL_GENERATOR);
+      condition.setWidthFull();
+      condition.setLabel("Condition");
+      condition.addClassName("combo-box");
+      condition.addClassName("chip-badge");
+      condition.setAllowCustomValue(false);
+    }
+
+    private void configureNumberFied() {
+      sampleSize.addClassName("number-field");
+      sampleSize.setLabel("Biological Replicates");
+      sampleSize.setStepButtonsVisible(true);
+      sampleSize.setStep(1);
+      sampleSize.setMin(1);
+      sampleSize.setValue(1.0);
+      sampleSize.setErrorMessage("Please specify a valid number of replicates");
+    }
+
+    void setAvailableVariableLevels(Collection<VariableLevel> availableVariableLevels) {
+      condition.setItems(availableVariableLevels);
+    }
+
+    void setCondition(Collection<VariableLevel> condition) {
+      this.condition.setValue(condition);
+    }
+
+    void setSampleSize(int size) {
+      sampleSize.setValue((double) size);
+    }
+
+    void registerForRemoveEvents(ComponentEventListener<RemoveEvent> listener) {
+      removeEventListeners.add(listener);
+    }
```

Comment:
I think having a consistent naming scheme for the listener methods would be beneficial e.g.: 
addDeletionEventListener()